src/main/java/com/example/springboot/dao/BookDAOService.java

```diff
@@ -0,0 +1,162 @@
+package com.example.springboot.dao;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Repository;
+
+import com.example.springboot.entity.Book;
+import com.mysql.jdbc.Statement;
+
+/**
+ * This class implement methods from IBookDAOService interface
+ * @author Kohut Dmytro
+ * @version 1.0
+ */
+@Repository("iBookService")
+public class BookDAOService implements IBookDAOService {
+
+	private final Logger LOG = Logger.getLogger(BookDAOService.class);
+	
+	private static final String SQL_FIND_ALL = "SELECT id, title, author, reader_id FROM `books`;";
+	private static final String SQL_SELECT_BY_ID = "SELECT id, title, author, reader_id FROM `books` WHERE id=?;";
+	private static final String SQL_CREATE = "INSERT INTO `books`(title, author, reader_id) VALUES(?, ?, ?);";
+	private static final String SQL_UPDATE = "UPDATE `books` SET title=?, author=?, reader_id=? WHERE id=?;";
+	private static final String SQL_DELETE = "DELETE FROM `books` WHERE id=?;";
+	
+	/**
+	 * This method create and return a connection
+	 * @return Connection
+	 * @throws SQLException 
+	 */
+	private Connection getConnection() throws SQLException {
+		return DriverManager.getConnection("jdbc:mysql://localhost:3306/libriary", "root", "root");
+	}
+	
+	/**
+	 * This method return all books from database
+	 * @return Collection<Book>
+	 */
+	@Override
+	public Collection<Book> findAll() {
+		List<Book> resultList = new ArrayList<>();
+		
+		try (Connection connection = getConnection()) {
+			PreparedStatement statement = connection.prepareStatement(SQL_FIND_ALL);
+			ResultSet resultSet = statement.executeQuery();
+			
+			while(resultSet.next()) {
+				Book book = new Book(resultSet.getInt("id"), resultSet.getString("title"),
```

Comment:
it is not the best way of using a constructor. Pls move into private method filling of those records